src/charts/donut/donut-percent-chart-directive.js

```diff
@@ -68,6 +133,16 @@ angular.module('patternfly.charts').directive('pfDonutPctChart', ['ChartsMixin',
           scope.config.donutConfig.data = scope.getDonutData(scope);
           setupDonutChartTitle();
         });
+        attrs.$observe('donutConfig', function() {
```

Comment:
Why not use `$scope.$watch()` here?  If this only happens once then you should be able to unbind the watcher after it happens and thus prevent an infinite loop.